autoprecompiles/src/trace_handler.rs

```diff
@@ -103,11 +106,16 @@ where
                 .iter()
                 .zip_eq(original_instruction_table_offsets.iter())
                 .map(|(air_id, dummy_table_offset)| {
-                    let Trace { values, width } = air_id_to_dummy_trace.get(air_id).unwrap();
+                    let trace = air_id_to_dummy_trace.get(air_id).unwrap();
+                    let values = trace.values();
+                    let width = trace.width();
                     let occurrences_per_record = air_id_occurrences.get(air_id).unwrap();
                     let start = (trace_row * occurrences_per_record + dummy_table_offset) * width;
-                    let end = start + width;
-                    &values[start..end]
+                    OriginalRowReference {
+                        data: values,
+                        start,
+                        length: width,
+                    }
```

Comment:
we keep the reference, start and end, because it's likely we'll pass them to GPU.